test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,153 @@
+              namespace: ns-3
+              pod: pod-1
+              phase: Failed
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-3 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Not Alerted cases:
+
+      # Pod is in the Unknown state and unscheduled (unscheduled once in the 15 min
+      # interval), so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-4", pod="pod-1"}'
+        values: '1x0 0x13'
+
+      # Pod is in Succeeded state and scheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Succeeded"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-5", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Running state for the first 14 min, then it's in the Failed state
+      # for the last 1 min and unscheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-7", phase="Running"}'
+        values: '1x13 0'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-7", phase="Failed"}'
+        values: '0x13 1'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-7", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Pending state and scheduled, but it has a namespace that starts
+      # with 'openshift' so it's ignored.
+      - series: 'kube_pod_status_phase{pod="pod-2", namespace="openshift-prod-test", phase="Pending"}'
+        values: '1x15'
+      - series: 'kube_pod_status_unschedulable{namespace="openshift-prod-test", pod="pod-2"}'
+        values: '0x15'
+
+      # Pod is in the Failed state and scheduled, but it has a namespace that ends
+      # with 'tenant' so it's ignored.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="kube-tenant", phase="Failed"}'
```

Comment:
This one both starts with kube and ends with tenant so we can't tell which one is it not alerting for.
Let's change it to something that does not start with kube, and add additional test cases to cover the rest of that regex (i.e. one with kube and one with default)

Reply:
Added all the cases.